tests/vmi_multus_test.go

```diff
@@ -875,6 +876,56 @@ var _ = Describe("[Serial]SRIOV", func() {
 				return libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA))
 			}, 15*time.Second, time.Second).Should(Succeed())
 		})
+
+		Context("With VLAN", func() {
+			var (
+				mac1 string
+				mac2 string
+				mac3 string
+			)
+
+			const (
+				cidrVlaned1          = "192.168.0.1/24"
+				cidrNonVlaned        = "192.168.0.3/24"
+				cidrVlaned2          = "192.168.0.2/24"
+				sriovVlanNetworkName = "sriov-vlan"
+			)
+
+			generateRandomMacAsString := func() string {
+				mac, err := tests.GenerateRandomMac()
+				Expect(err).ToNot(HaveOccurred())
+				return mac.String()
+			}
+
+			BeforeEach(func() {
+				createNetworkAttachementDefinition(sriovVlanNetworkName, tests.NamespaceTestDefault, sriovConfVlanCRD)
+				mac1 = generateRandomMacAsString()
+				mac2 = generateRandomMacAsString()
+				mac3 = generateRandomMacAsString()
+			})
+
+			It("should be able to ping between two VMIs with the same VLAN over SRIOV network", func() {
+				// create an anonymous vmi to be used for pinging
+				createSriovVMIWithStaticIPByMac(sriovVlanNetworkName, mac1, cidrVlaned1)
+				vlanedVMI := createSriovVMIWithStaticIPByMac(sriovVlanNetworkName, mac2, cidrVlaned2)
+
+				By("pinging between vlanedVMI1 and vlanedVMI2 over vlan")
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vlanedVMI, cidrToIP(cidrVlaned1))
+				}, 15*time.Second, time.Second).ShouldNot(HaveOccurred())
+			})
+
+			It("should NOT be able to ping between Vlaned VMI and a non Vlaned VMI", func() {
+				// create an anonymous vmi to be used for pinging
+				createSriovVMIWithStaticIPByMac(sriovVlanNetworkName, mac1, cidrVlaned1)
+				nonVlanedVMI := createSriovVMIWithStaticIPByMac(sriovnet3, mac3, cidrNonVlaned)
```

Comment:
same

Reply:
Done.
Modified the exiting CreateSiorvVms and use it, were the waiting is concurrent .